src/main/java/edu/harvard/h2ms/service/utils/H2msRestUtils.java

```diff
@@ -0,0 +1,108 @@
+
+    /**
+     * Iterates through a list of events and parses the timestamps specified
+     * by a timeframe.
+     * Ex. If the specified timeframe is 'month', the timestamp is parsed as follows:
+     *      "2018-03-21T17:58:05.742+0000"  --> "March (2018)"
+     * @param events
+     * @param timeframe
+     * @return
+     */
+    public static List<String> extractParsedTimestamps(List<Event> events, String timeframe){
+        if(events.size() ==  0) {
+            log.warn("There are 0 events in the H2MS system.");
+            return new ArrayList<>();
+        }
+        // Parses event timestamps based on timeframe type
+        List<String> values = new ArrayList<>();
+        for (Event event : events) {
+            Calendar cal = Calendar.getInstance();
+            cal.setTime(event.getTimestamp());
+            if(timeframe.equals("week")) {
+                values.add(formatWeek(cal.get(Calendar.WEEK_OF_YEAR), cal.get(Calendar.YEAR)));
+            } else if (timeframe.equals("month")) {
+                values.add(formatMonth(cal.get(Calendar.MONTH), cal.get(Calendar.YEAR)));
+            } else if (timeframe.equals("year")) {
+                values.add(String.valueOf(cal.get(Calendar.YEAR)));
+            } else if (timeframe.equals("quarter")) {
+                values.add(formatQuarter(cal.get(Calendar.MONTH), cal.get(Calendar.YEAR)));
+            }
+        }
+        return values;
+    }
+
+    public static Map<String, Long> frequencyCounter (List<String> parsedTimestamps){
+        return parsedTimestamps.stream().collect(Collectors.groupingBy(x -> x, Collectors.counting()));
+    }
+
+    private static String formatMonth(Integer month, Integer year){
+        if(month.equals(0)){
+            return "January " + "(" + year + ")";
+        } else if(month.equals(1)){
+            return "February " + "(" + year + ")";
+        } else if(month.equals(2)){
+            return "March " + "(" + year + ")";
+        } else if(month.equals(3)){
+            return "April " + "(" + year + ")";
+        } else if(month.equals(4)){
+            return "May " + "(" + year + ")";
+        } else if(month.equals(5)){
+            return "June " + "(" + year + ")";
+        } else if(month.equals(6)){
+            return "July " + "(" + year + ")";
+        } else if(month.equals(7)){
+            return "August " + "(" + year + ")";
+        } else if(month.equals(8)){
+            return "September " + "(" + year + ")";
+        } else if(month.equals(9)){
+            return "October " + "(" + year + ")";
+        } else if(month.equals(10)){
+            return "November " + "(" + year + ")";
+        } else if(month.equals(11)){
+            return "December " + "(" + year + ")";
+        } else {
+            return "";
+        }
+
+    }
+
+    public static String formatWeek(Integer week, Integer year){
+        // Checks the digits in the ones places
+        char digit = week.toString().charAt(1);
```

Comment:
Throws a java.lang.StringIndexOutOfBoundsException when you use week "1". Since charAt is trying to get the second digit, which weeks 1-9 do not have.

Reply:
Please add a test to expose this issue, then code until the test passes.